tests/drivers/uart/uart_async_api/Kconfig

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2024 STMicroelectronics
+#
+# SPDX-License-Identifier: Apache-2.0
+
+mainmenu "UART Async Test"
+
+source "Kconfig.zephyr"
+
+if DCACHE
+
+config DT_DEFINED_NOCACHE
```

Comment:
Perhaps it's time to generalize the DT NOCACHE? Considering its usage here, in the SPI tests, and potentially by applications.

For example, by introducing related configs in https://github.com/zephyrproject-rtos/zephyr/blob/414b3b7efd3e927a7e68813217285b30a160d4b2/arch/Kconfig#L385

Reply:
> Perhaps it's time to generalize the DT NOCACHE? Considering its usage here, in the SPI tests, and potentially by applications.

@GeorgeCGV Will do it in another PR for all drivers as soon as this one is merged.